scrape/scrape.go

```diff
@@ -80,18 +82,18 @@ type FailureLogger interface {
 
 // scrapePool manages scrapes for sets of targets.
 type scrapePool struct {
-	appendable storage.Appendable
-	logger     *slog.Logger
-	cancel     context.CancelFunc
-	httpOpts   []config_util.HTTPClientOption
+	ctx    context.Context
+	cancel context.CancelFunc
+	logger *slog.Logger
+
+	httpOpts []config_util.HTTPClientOption
```

Comment:
This seems like configuration option that's shared accross loops and I don't think this can change either, at least I don't see assignment for it, so maybe should be in `scrapeLoopOptions` ?